Algorithm/QCAlgorithm.History.cs

```diff
@@ -51,6 +51,11 @@ public bool IsWarmingUp
         /// <param name="timeSpan">The amount of time to warm up, this does not take into account market hours/weekends</param>
         public void SetWarmup(TimeSpan timeSpan)
         {
+            if (!IsWarmingUp)
```

Comment:
Cool!
I think more precisely we could use the algorithms `_locked` field, which is false until after being initialized, seems more exact for the behavior we want? Think calling `SetWarmup` during warmup should also throw.

Maybe we could make this method directly call `SetWarmup(TimeSpan timeSpan, Resolution resolution)`. With resolution null. Same for `public void SetWarmup(int barCount)` -> `public void SetWarmup(int barCount, Resolution resolution)`

Reply:
Gotcha!